source/AI.cpp

```diff
@@ -89,12 +89,17 @@ namespace {
 	// ship. Carried escorts are waited for in AI::Step.
 	bool EscortsReadyToJump(const Ship &ship)
 	{
-		for(const weak_ptr<Ship> &escort : ship.GetEscorts())
+		bool shipIsYours = ship.IsYours();
+		const Government *gov = ship.GetGovernment();
+		for(const weak_ptr<Ship> &ptr : ship.GetEscorts())
 		{
-			shared_ptr<const Ship> locked = escort.lock();
-			if(locked && !locked->IsDisabled() && !locked->CanBeCarried()
-					&& locked->GetSystem() == ship.GetSystem()
-					&& locked->JumpFuel() && !locked->IsReadyToJump(true))
+			shared_ptr<const Ship> escort = ptr.lock();
+			// Skip escorts which are not player-owned and not escort mission NPCs.
+			if(!escort || gov->IsEnemy(escort->GetGovernment()) || (shipIsYours && !escort->IsYours() && !escort->GetPersonality().IsEscort()))
```

Comment:
```suggestion
			if(!escort || (shipIsYours && !escort->IsYours() && (!escort->GetPersonality().IsEscort() || gov->IsEnemy(escort->GetGovernment()))))
```

I believe it should be like this.

Reply:
My original change should apply to NPCs as well.  An NPC shouldn't consider a hostile escort an escort.

Reply:
Is it possible for NPCs to have an escort that is hostile?

Reply:
I don't think it is.

Reply:
I mean that makes sense; usually hostile escorts _I think_ are the result of player human mistakes.  @quyykk then do you still want rhis change?  If so I'll be able to commit it later in about 10h.